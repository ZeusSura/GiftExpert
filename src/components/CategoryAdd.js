import React,{useState} from 'react'
import PropTypes from 'prop-types'

const CategoryAdd = ({setCategorias}) => {

    const [inputValue,setInputValue] = useState('')
    const handleInput = e =>{
        setInputValue(e.target.value)
    }
    const handleSubmit  = e=>
    {        
        e.preventDefault()
        if(inputValue.trim()==='')
        {
            return
        }
        setCategorias(categorias=>[inputValue,...categorias])
        setInputValue('')
    }

    return (
        
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
            id="categoria"
            name="categoria"
            className="input-categoria"
            type="text"
            value={inputValue}
            onChange={handleInput}
            />
        </form>
    )
}


CategoryAdd.propTypes = {

    setCategorias:PropTypes.func.isRequired
}
export default CategoryAdd ;