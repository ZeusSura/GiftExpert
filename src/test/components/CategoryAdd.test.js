import React from "react";
import CategoryAdd from "../../components/CategoryAdd";
import { shallow } from "enzyme";

describe("Pruebas de categoria add", () => {
  const setCategorias =jest.fn();
  let wraper = shallow(<CategoryAdd setCategorias={setCategorias} />);
  beforeEach(()=>{
    jest.clearAllMocks()
    wraper = shallow(<CategoryAdd setCategorias={setCategorias} />);
  })
  
  test("pruebas de snapshot", () => {
    expect(wraper).toMatchSnapshot();
  });
  test("Haciendo pruebas sobre el input", () => {
    const input =  wraper.find('.input-categoria')
    const value = "Este es un valor"
    input.simulate('change',{target:{value}})
    
    expect(wraper.find('p').text().trim()).toBe(value)
  });

  test("No debe postear la informacion on submite",()=>{

    wraper.find('form').simulate('submit',{preventDefault(){}})
    expect(setCategorias).not.toHaveBeenCalled()
  })

  test("simular como enviaria datos normalmente",()=>{
    //Se crea el formulario
    const value = "Este es un valor"
    const input =  wraper.find('.input-categoria')
    input.simulate('change',{target:{value}})
    //Se locailiza el formulario y envia los datos
    wraper.find('form').simulate('submit',{preventDefault(){}})
    expect(setCategorias).toHaveBeenCalled()
    expect(input.props().value.trim()).toBe('')
  });
});
