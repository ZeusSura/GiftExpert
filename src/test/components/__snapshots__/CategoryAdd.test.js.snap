// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de categoria add pruebas de snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    className="input-categoria"
    id="categoria"
    name="categoria"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
